plugins/field-bitmap/src/field-bitmap.js

```diff
@@ -0,0 +1,464 @@
+        this.bindEvent_(button, 'mouseenter', () => {
+          this.onMouseEnterPixel_(r, c);
+        });
+      }
+      pixelContainer.appendChild(rowDiv);
+    }
+
+    // Add control buttons below the pixel grid
+    this.addControlButton_(dropdownEditor, 'Randomize', this.randomizePixels_);
+    this.addControlButton_(dropdownEditor, 'Clear', this.clearPixels_);
+
+    this.render_();
+
+    return dropdownEditor;
+  }
+
+  /**
+   * Initializes the on-block display.
+   * @override
+   */
+  initView() {
+    this.blockDisplayPixels_ = [];
+    for (let r = 0; r < this.imgHeight_; r++) {
+      const row = [];
+      for (let c = 0; c < this.imgWidth_; c++) {
+        const square = Blockly.utils.dom.createSvgElement(
+            'rect', {
+              'x': c * PIXEL_SIZE,
+              'y': r * PIXEL_SIZE,
+              'width': PIXEL_SIZE,
+              'height': PIXEL_SIZE,
+              'fill': EMPTY_PIXEL_COLOR,
+              'fill_opacity': 1,
+            },
+            this.fieldGroup_);
+        row.push(square);
+      }
+      this.blockDisplayPixels_.push(row);
+    }
+  }
+
+  /**
+   * Updates the size of the block based on the size of the underlying image.
+   * @override
+   * @protected
+   */
+  updateSize_() {
+    {
+      const newWidth = PIXEL_SIZE * this.imgWidth_;
+      const newHeight = PIXEL_SIZE * this.imgHeight_;
+      if (this.borderRect_) {
+        this.borderRect_.setAttribute('width', String(newWidth));
+        this.borderRect_.setAttribute('height', String(newHeight));
+      }
+
+      this.size_.width = newWidth;
+      this.size_.height = newHeight;
+    }
+  }
+
+  /**
+   * Creates a button with specified text and action, and adds it to the
+   * parent view.
+   */
+  addControlButton_(parent, buttonText, onClick) {
+    const button = this.createElementWithClassname_('button', 'controlButton');
+    button.innerHTML = buttonText;
+    parent.appendChild(button);
+    this.bindEvent_(button, 'click', onClick);
+  }
+
+  /**
+   * Disposes of events belonging to the slider editor.
+   * @private
+   */
+  dropdownDispose_() {
+    for (const event of this.boundEvents_) {
+      Blockly.browserEvents.unbind(event);
+    }
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Constructs an array of zeros with the specified width and height
+   * @return the new value
+   */
+  getEmptyArray_() {
+    const newVal = [];
+    for (let r = 0; r < this.imgWidth_; r++) {
+      newVal.push([]);
+      for (let c = 0; c < this.imgHeight_; c++) {
+        newVal[r].push(0);
+      }
+    }
+    return newVal;
+  }
+
+  /**
+   * Called when a mousedown event occurs within the bounds of a pixel.
+   * @private
+   */
+  onMouseDownInPixel_(r, c) {
+    // Toggle that pixel to the opposite of its value
+    const newPixelValue = 1 - this.getValue()[r][c];
+    this.setPixel_(r, c, newPixelValue);
+    this.mouseIsDown_ = true;
+    this.valToPaintWith_ = newPixelValue;
```

Comment:
I like using the new value to set the paint color. I'm obvs not UX, but seems very intuitive!